test/cluster/config_test.go

```diff
@@ -0,0 +1,419 @@
+		return cert, err
+	}
+	certBytes, err := os.ReadFile(userCertPath)
+	if err != nil {
+		return cert, err
+	}
+	certKeyPair, err := tls.X509KeyPair(certBytes, keyBytes)
+	if err != nil {
+		return cert, err
+	}
+
+	return certKeyPair, err
+}
+
+func updateServersAdmin(newAdmin string, newAdminKeyPath string, newAdminCertPath string, c *setup.Cluster) error {
+	for _, s := range c.Servers {
+		s.SetAdminID(newAdmin)
+		s.SetAdminKeyPath(newAdminKeyPath)
+		s.SetAdminCertPath(newAdminCertPath)
+		newAdminSigner, err := crypto.NewSigner(
+			&crypto.SignerOptions{KeyFilePath: newAdminKeyPath},
+		)
+		if err != nil {
+			return err
+		}
+		s.SetAdminSigner(newAdminSigner)
+	}
+	return nil
+}
+
+// Scenario:
+// - add alice to the admins' list
+// - add new user newAdmin to the admins' list
+// - trying to add alice again to the admins' list => the tx fails because alice is already an admin
+func TestAddNewAdmins(t *testing.T) {
+	dir, err := ioutil.TempDir("", "int-test")
+	require.NoError(t, err)
+
+	nPort, pPort := getPorts(1)
+	setupConfig := &setup.Config{
+		NumberOfServers:     1,
+		TestDirAbsolutePath: dir,
+		BDBBinaryPath:       "../../bin/bdb",
+		CmdTimeout:          10 * time.Second,
+		BaseNodePort:        nPort,
+		BasePeerPort:        pPort,
+		CheckRedirectFunc: func(req *http.Request, via []*http.Request) error {
+			return errors.Errorf("Redirect blocked in test client: url: '%s', referrer: '%s', #via: %d", req.URL, req.Referer(), len(via))
+		},
+	}
+	c, err := setup.NewCluster(setupConfig)
+	require.NoError(t, err)
+	defer c.ShutdownAndCleanup()
+
+	require.NoError(t, c.Start())
+
+	leaderIndex := -1
+	require.Eventually(t, func() bool {
+		leaderIndex = c.AgreedLeader(t, 0)
+		return leaderIndex >= 0
+	}, 30*time.Second, 100*time.Millisecond)
+	leaderServer := c.Servers[leaderIndex]
+
+	//alice is already a user in the cluster
+	certKeyPair, err := getUserCert("alice", c)
+	require.NoError(t, err)
+	require.NotNil(t, certKeyPair)
+
+	alice := []*types.Admin{
+		{
+			Id:          "alice",
+			Certificate: certKeyPair.Certificate[0],
+		},
+	}
+
+	//get current cluster config
+	configEnv, err := leaderServer.QueryConfig(t)
+	require.NoError(t, err)
+	require.NotNil(t, configEnv)
+	newConfig := configEnv.GetResponse().GetConfig()
+
+	//add alice to admins list
+	newConfig.Admins = append(newConfig.Admins, alice...)
+	version := configEnv.GetResponse().GetMetadata().GetVersion()
+
+	txID, rcpt, err := leaderServer.SetConfigTx(t, newConfig, version, c.Servers[leaderIndex].AdminSigner(), "admin")
+	require.NoError(t, err)
+	require.NotNil(t, rcpt)
+	require.True(t, txID != "")
+	require.True(t, len(rcpt.GetHeader().GetValidationInfo()) > 0)
+	require.Equal(t, types.Flag_VALID, rcpt.Header.ValidationInfo[rcpt.TxIndex].Flag)
+	t.Logf("tx submitted: %s, %+v", txID, rcpt)
+
+	//add new user - newAdmin
+	err = c.CreateAdditionalUserCryptoMaterials("newAdmin")
+	require.NoError(t, err)
+	certKeyPair, err = getUserCert("newAdmin", c)
```

Comment:
you have `bob` and `charlie` ready to use, like alice, no need to generate another one. Line 80 `cluster_setup.go`. If you need more, add `david`....